frontend/packages/knative-plugin/src/utils/create-eventsources-utils.ts

```diff
@@ -264,190 +248,59 @@ export const sanitizeKafkaSourceResource = (formData: EventSourceFormData): Even
   };
 };
 
-export const getEventSourceConnectorList = (
-  namespace: string,
-  csvData: ClusterServiceVersionKind[],
-) =>
-  _.reduce(
-    csvData,
-    (acm, cv) => {
-      const parseCSVData = parseALMExamples(cv, false);
-      _.forEach(parseCSVData, (res) => {
-        if (
-          _.findIndex(acm, res) === -1 &&
-          referenceFor(res) === referenceForModel(EventSourceCamelModel) &&
-          res?.metadata?.labels?.[EVENT_SOURCE_LABEL] === 'true'
-        ) {
-          const { apiGroup, plural, kind } = modelFor(referenceFor(res));
-          const {
-            metadata: { name },
-          } = res;
-          const modelData = {
-            [name]: {
-              name,
-              iconUrl: getEventSourceIcon(kind, res),
-              displayName: _.startCase(name),
-              title: _.startCase(name),
-              data: { almData: res },
-            },
-          };
-          acm.push(
-            checkAccess({
-              group: apiGroup,
-              resource: plural,
-              namespace,
-              verb: 'create',
-            }).then((result) => (result.status.allowed ? modelData : {})),
-          );
-        }
-      });
-      return acm;
-    },
-    [],
-  );
+export const getKameletMetadata = (kamelet: K8sResourceKind): EventSourceMetaData => {
+  let normalizedKamelet = {};
+  if (kamelet?.kind === CamelKameletModel.kind) {
+    const {
+      kind,
+      metadata: { annotations },
+      spec: {
+        definition: { title, description },
+      },
+    } = kamelet;
+    const provider = annotations[CAMEL_K_PROVIDER_ANNOTATION];
+    const iconUrl = getEventSourceIcon(kind, kamelet);
+    normalizedKamelet = {
+      name: title,
+      description,
+      provider,
+      iconUrl,
+    };
+  }
+  return normalizedKamelet as EventSourceMetaData;
+};
 
-export const getEventSourceList = (namespace: string, eventSourceModels: K8sKind[]) => {
+export const getEventSourceMetadata = (eventSourceModel: K8sKind, t): EventSourceMetaData => {
+  let normalizedSource = {};
+  if (eventSourceModel) {
+    const { kind, label: name } = eventSourceModel;
+    const { description, provider } = getEventSourceCatalogProviderData(kind, t) ?? {};
+    normalizedSource = {
+      name,
+      description,
+      provider,
+      iconUrl: getEventSourceIcon(referenceForModel(eventSourceModel)),
+    };
+  }
+  return normalizedSource as EventSourceMetaData;
+};
+
+export const getEventSourceModelsWithAccess = (namespace: string, eventSourceModels: K8sKind[]) => {
```

Comment:
Why not return the map itself here, instead of pushing to accessList?
```
return eventSourceModels.map((model) => {
    const { apiGroup, plural } = model;
   return checkAccess({
                group: apiGroup,
        	resource: plural,
        	namespace,
   	        verb: 'create',
	      }).then((result) => (result.status.allowed ? model : null)),